### PR TITLE
ddl: args v2 for create/drop/modify schema

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/53930

Problem Summary:

### What changed and how does it work?
- args v2 for `create-schema`/`drop-schema`/`modify schema charset or collate` / `modify schema placement`
- remove ArgsV2/RawArgsV2, we can reuse Args/RawArgs actually
- add `FinishedJobArgs` interface for jobs that need to save some info after job finish
- add some senity check

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/meta/model/job.go [hunk 1]
```diff
@@ -512,11 +513,15 @@ func (job *Job) Encode(updateRawArgs bool) ([]byte, error) {
 				}
 			}
 		} else {
-			job.RawArgsV2, err = json.Marshal(job.ArgsV2)
+			var arg any
+			if len(job.Args) > 0 {
+				arg = job.Args[0]
+			}
+			job.RawArgs, err = json.Marshal(arg)
```
Comment (lance6716) [inline pkg/meta/model/job.go:520]: if can moved inside `if len(job.Args) > 0`? And do you think we should assert `len(job.Args) == 1`

File: pkg/meta/model/job_args.go [hunk 2]
```diff
@@ -41,6 +43,146 @@ type JobArgs interface {
 	fillJob(job *Job)
 }
 
+// FinishedJobArgs is the interface for finished job arguments.
+// in most cases, job args are cleared out after the job is finished, but some jobs
+// will write some args back to the job for other components.
+type FinishedJobArgs interface {
```
Comment (lance6716) [inline pkg/meta/model/job_args.go:49]:
```suggestion
type finishedJobArgs interface {
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: pkg/meta/model/job.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/meta/model/job.go:520]: no, we need Marshal nil to clear args out

> And do you think we should assert len(job.Args) == 1

add in another PR, https://github.com/pingcap/tidb/pull/55964/files#diff-c51ad6b394f865592721a3e18d58574176394570e3efcff7703e8bd0c3c5c2efR518

File: pkg/meta/model/job_args.go (hunk 2 above)
Comment (D3Hunter) [reply pkg/meta/model/job_args.go:49]: it's used in some test, see https://github.com/pingcap/tidb/pull/55919/files#diff-90a20211c2de727da7d84a8bb29745bf631f43fa1a30394e4f60adfadcd7276cR727

**Review by lance6716 — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**